### ISSUE TITLE
Mask is being ignored when cudnn_flash_attention is used

### ISSUE THREAD

**Issue by finbarrtimbers** [AI-Hypercomputer/maxtext#878]
From the [TransformerEngine docs](https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/api/jax.html#transformer_engine.jax.flax.DotProductAttention):

> mask in [__call__](https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/api/jax.html#transformer_engine.jax.flax.DotProductAttention.__call__) is ignored for ‘no_mask’ and ‘causal’.

I think that `attn_mask_type` should be set to `causal_padding`, or else it'll ignore the mask being passed on line 370.

**Comment by bvandermoon** [comment]
Thanks Finbarr. Created a [PR to address this](https://github.com/AI-Hypercomputer/maxtext/pull/913)

**Comment by bvandermoon** [comment]
Resolving this issue since the PR has been merged. Please reopen or let us know if you have any further issues
